File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        if len(nums) <= 1: return nums
        mid = math.floor(len(nums)/2)
        return Solution.merge(self, Solution.sortArray(self, nums[0:mid]), Solution.sortArray(self, nums[mid:]))

    def merge(self, left, right):
        res = []
        while 0 < len(left) and 0 < len(right):
            if left[0] <= right[0]:
                res.append(left[0])
                del left[0]
            else:
                res.append(right[0])
                del right[0]
        if len(left) > 0:
            res = res + left
        else:
            res = res + right
        return res

File: 0912-sort-an-array/0912-sort-an-array.cpp
class Solution {
public:
    vector<int> sortArray(vector<int>& nums) {
            if (nums.size() <= 1) return nums;
            vector<int> left (nums.begin(), nums.begin() + nums.size()/2);
            vector<int> right (nums.begin() + nums.size() / 2, nums.end());
            return merge(sortArray(left), sortArray(right));
        };

    vector<int> merge(vector<int>left, vector<int>right) {
        vector<int> result;
        auto left_it = left.begin();
        auto right_it = right.begin();

        while (left_it != left.end() && right_it != right.end()) {
            if (*left_it <= *right_it) {
                result.push_back(*left_it);
                ++left_it;
            }
            else {
                result.push_back(*right_it);
                ++right_it;
            };
        };

        if (left_it != left.end()) {
            result.insert(result.end(), left_it, left.end());
        }
        else {
            result.insert(result.end(), right_it, right.end());
        };
        return result;
    };
};